src/hooks/useBanner.ts

```diff
@@ -0,0 +1,141 @@
+import { useMemo } from 'react';
+import DOMPurify from 'dompurify';
+import yaml from 'js-yaml';
+import { DateTime } from 'luxon';
+import { marked } from 'marked';
+import { useK8sWatchResource } from '~/k8s';
+import { ConfigMap } from '~/types/configmap';
+import { bannerConfigYupSchema } from '~/utils/validation-utils';
+import { ConfigMapGroupVersionKind, ConfigMapModel } from './../models/config-map';
+
+export type BannerType = 'info' | 'warning' | 'danger';
+export type BannerConfig = {
+  enable: boolean;
+  summary: string;
+  type: BannerType;
+  year?: string;
+  month?: string;
+  startTime?: string;
+  endTime?: string;
+  timeZone?: string;
+  repeatType?: 'none' | 'weekly' | 'monthly';
+  dayOfWeek?: number; // 0-6 for Sunday-Saturday
+  dayOfMonth?: number; // 1-31 for the day of the month
+};
+
+function convertToTimeZone(date: Date, timeZone: string): Date {
+  return DateTime.fromJSDate(date).setZone(timeZone).toJSDate();
+}
+
+// Fore repeated banners
+function parseHM(hm: string): number {
+  const [h, m] = hm.split(':').map(Number);
+  return h * 60 + m;
+}
+
+const parseBanner = (yamlContent: string): BannerConfig | null => {
+  const parsed = yaml.load(yamlContent, { schema: yaml.FAILSAFE_SCHEMA });
+
+  if (typeof parsed !== 'object' || parsed === null) return null;
+  try {
```

Comment:
@sahil143  

Thanks for raising this!
I currently use pure YAML in the ConfigMap for several reasons:

    YAML supports comments, which helps document the meaning of each field and improves maintainability.
    YAML is widely used in Kubernetes ecosystem and is familiar to our operators.

The structure of the banner config is a bit complex , and YAML’s flexibility helps express that clearly and human readable for anyone who would like to set banners.

In our codes, YAML is pretty easy to convert as json and do not bring unexpected complexity .

So, I would like to keep yaml and leave the easiest way to any possible users.

Reply:
> The structure of the banner config is a bit complex , and YAML’s flexibility helps express that clearly and human readable for anyone who would like to set banners.

What do you mean by YAML's flexibility? JSON provides the same flexibility. Also for UI related config maps we use JSON in config maps in `konflux-info`. For consistency, I would prefer we use JSON for this use case also.
As for your concern around comments. You have added a JSON schema and really good docs explaining the usage of banners. Using yaml just add another overhead on the UI to parse the YAMl to JSON.